server/config.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"errors"
+	"time"
+)
+
+type ShipperServerConfig struct {
+	// PollingInterval is an interval for polling queue metrics and updating the indices.
+	// Must be greater than 0
+	PollingInterval time.Duration
```

Comment:
Continuing from my earlier comment on this interacting with the Filebeat `registry.flush` parameter, I wonder if this should be a parameter in `PersistedIndexRequest` instead of the configuration?

Controlling this through the configuration means that we will have to expose it to the user in the agent policy output settings where the shipper gets its configuration from. It feels like it would be difficult to get users to understand what it controls there as the effects of this parameter affect inputs more than outputs.


Making it a parameter of `PersistedIndexRequest` puts the choice on the individual input developers as to how they want to use it. For Filebeat this means it is controlled through the already documented `registry.flush` setting.

We can then either document a reasonable default value like `1s` or just enforce one through the API if this parameter isn't specified.

Reply:
@cmacknz what you're suggesting here cannot be implemented without switching to timeouts per request or I'm missing something?

The only way to have it as a part of the request is to run a timer per subscriber like @faec suggested.

So, my question is: did we already commit to the timer per connection solution and I should re-implement everything accordingly?

Reply:
After talking face to face we agreed to go forward with the timer approach instead of centralised conditional variable, so the requested feature is now addressed with this PR

https://github.com/elastic/elastic-agent-shipper-client/pull/12